x-pack/plugins/rule_registry/server/rule_data_plugin_service/resource_installer.ts

```diff
@@ -252,10 +256,24 @@ export class ResourceInstaller {
     indexInfo: IndexInfo,
     namespace: string
   ): Promise<void> {
-    const { logger } = this.options;
+    const { logger, frameworkAlerts } = this.options;
 
     const alias = indexInfo.getPrimaryAlias(namespace);
 
+    if (namespace === 'default' && frameworkAlerts.enabled()) {
+      const initialized = await frameworkAlerts.isContextInitialized(
+        indexInfo.indexOptions.registrationContext
+      );
+
+      if (!initialized) {
```

Comment:
Framework AAD resources are installed on plugin setup so if there is an error, it will be thrown and logged by the alerting framework. Throwing this error here if the resources are not installed ensures that rule registry writing is disabled which matches current rule registry behavior.

Reply:
Done in https://github.com/elastic/kibana/pull/151792/commits/f18446a89f78bdad8119cd8b786612affc29f505